### DISCUSSION TITLE
SET_ORIGIN_AT_HOME_POS and Soft Limits

### DISCUSSION THREAD

**Discussion by mvackel** [Paciente8159/uCNC#540] [Config and issues] [answered]
After compiling with the option "SET_ORIGIN_AT_HOME_POS" (in "cnc_config.h" the jog controls are not working. 
I set this option to zero the machine coordinates after homing (because of inverted limit switches, $23). This way, all machine coordinates will be negative (as suggested in many places). This is working ok, as after homing UGS displays correctly all coordinates as zero (... shouldn't be the negative homing pull-off ($27), as in GRBL?). 

The problem is that the jog commands do not work if the soft limits ($20) are enabled. If set $20 to zero, disabling soft limits, jog commands work normally. It seems that uCNC doens't allow negative machine coordinates. Shouldn't this be compatible with GRBL, allowing negative machine coordinates?
If this option is disable (as the default, with #define commented), the machine homing sets the axis max travel limits specified ($13x) and soft limits and jogs work normally.

The comment in file "cnc_config.h" related to this #define seems to be compatible to GRBL but the soft limits are not working as expected. Maybe I'm missing something....

**Comment by Paciente8159** [comment, marked as answer]
This was just by looking at the code (I did not had a chance to test it yet), but I believe this is a bug caused by the logic in boundries check with option SET_ORIGIN_AT_HOME_POS. Try this please.

Open src/hal/kinematic/kinematic_cartesian.c and go to line [164](https://github.com/Paciente8159/uCNC/blob/f340878a3612d8a8b69933c6636c05de928c760c/uCNC/src/hal/kinematics/kinematic_cartesian.c#L164) .
Replace this:
`float value = !(g_settings.homing_dir_invert_mask & (1 << i)) ? -axis[i] : axis[i];`
By this
`float value = !(g_settings.homing_dir_invert_mask & (1 << i)) ? axis[i] : -axis[i];`

Please let me know if it fixes the issue.
Once again thank you for reporting this.

**Reply by mvackel** [reply]
It works! Changing this line fixes the issue.

**Reply by Paciente8159** [reply]
Thanks. This bug is also present at core xy kinematic. I must check the rest

**Reply by Paciente8159** [reply]
This as been patched in both master branch and on the v1.7.x-bugfix
